<launch>

  <!-- Input arguments -->
  <arg name="device_number" value="0" /> 
  <arg name="device" value="6843" doc="TI mmWave sensor device type [1443, 1642, 6843]"/>
  <arg name="config" value="3d" doc="TI mmWave sensor device configuration [3d_best_range_res (not supported by 1642 EVM), 2d_best_range_res]"/>
  <arg name="max_allowed_elevation_angle_deg" default="90" doc="Maximum allowed elevation angle in degrees for detected object data [0 > value >= 90]}"/>
  <arg name="max_allowed_azimuth_angle_deg" default="90" doc="Maximum allowed azimuth angle in degrees for detected object data [0 > value >= 90]}"/>
  <arg name="punti_desiderati" default="200"/>
  <arg name="use_agglo" default="True"/>
  <arg name="use_lior" default="false"/>
  <arg name="take_sideinfo" default="false"/>
  <arg name="dist_InlierTh" default="0.2"/>
  <arg name="cdist" default="0.5"/>

  <!-- mmWave_Manager node -->

  <node pkg="ti_mmwave_rospkg" type="ti_mmwave_rospkg" name="ti_mmwave_rx" ns="radar_rx" output="screen">
    <param name="device_number" value="$(arg device_number)"/>
    <param name="device_name" value="$(arg device_name)"   />
    <param name="command_port" value="/dev/$(arg device_model)_$(arg device_name)_COMMAND"  />
    <!--<param name="command_port" value="/dev/ttyUSB0"  />-->
    <param name="command_rate" value="115200"   />
    <param name="data_port" value="/dev/$(arg device_model)_$(arg device_name)_DATA"  />
    <param name="data_rate" value="921600"   />
    <param name="max_allowed_elevation_angle_deg" value="$(arg max_allowed_elevation_angle_deg)"   />
    <param name="max_allowed_azimuth_angle_deg" value="$(arg max_allowed_azimuth_angle_deg)"   />
    <param name="frame_id" value="turtlebot3/ti_mmwave_0"/>
    <remap from="/ti_mmwave/radar_scan_pcl" to="/ti_mmwave_rx/radar_scan_pcl"/>
    <param name="pdesiderati" value="$(arg punti_desiderati)"/>
    <param name="use_agglo" value="$(arg use_agglo)"/>
    <param name="use_lior" value="$(arg use_lior)"/>
    <param name="take_sideinfo" value="$(arg take_sideinfo)"/>
    <param name="distInlier" value="$(arg dist_InlierTh)"/>
    <param name="cdist" value="$(arg cdist)"/>
  </node>   


  <!-- mmWaveQuickConfig node (terminates after configuring mmWave sensor) -->
 
  <node pkg="ti_mmwave_rospkg" type="mmWaveQuickConfig" name="ti_mmwave_config_$(arg device_number)" ns="radar_rx" args="$(find ti_mmwave_rospkg)/cfg/optimalConfig6843.cfg" output="screen" />
  <node pkg="tf" type="static_transform_publisher" name="radar_baselink_0" args="0 0 0 0 0 0 /turtlebot3/base_link /turtlebot3/ti_mmwave_0 100"/>

</launch>
